flax/core/scope.py

```diff
@@ -14,13 +14,52 @@
 
 """Flax functional core: Scopes.
 
-The goal of the Flax functional core is to provide a purely functional 
-abstraction that takes care of various types of bookkeeping such as parameter
-and RNG management. The main abstraction is Scope, which contains parameters
-and RNGs, and which can be nested.
-
-TODO: elaborate.
-
+Flax deep learning models contain variables, which can be of different 
+categories such as trainable parameters, or internal state variables that aren't
+parameters (e.g., batch statistics). For instance, the top-level variable
+dictionary of a Flax model using batch normalization may look as follows::
+  {
+    "params": {
+      "Conv1": { "weight": ..., "bias": ... },
+      "BatchNorm1": { "scale": ..., "mean": ... },
+      "Conv2": {...}
+    },
+    "batch_stats": {
+      "BatchNorm1": { "moving_mean": ..., "moving_average": ...}
```

Comment:
nit: moving_mean and moving_average are the same thing. Normally we have ema_mean and ema_variance